### PR TITLE
adding captcha and rate limit for emails

### PR DESCRIPTION
**Pull request by Suvrat1629**
#120 added captcha and rate limit for emails

### CODE REVIEW

**Comment by Suvrat1629** [comment]
@bambooch updates?

**Comment by bambooch** [comment]
> @bambooch updates?
> 

@Suvrat1629 thanks for opening the PR!sorry still didn't get the time to properly review.

File: backend/controllers/email.controller.js [hunk 1]
```diff
@@ -2,54 +2,67 @@ const nodemailer = require("nodemailer");
 require("dotenv").config();
 
 class EmailController {
+  constructor() {
+    // Set up reCAPTCHA with your site key and secret key
+    this.recaptcha = new RecaptchaV2(process.env.RECAPTCHA_SITE_KEY, process.env.RECAPTCHA_SECRET_KEY);
```
Comment (bambooch) [inline backend/controllers/email.controller.js:7]: how exactly to you setup the env variable for this? can you add a reference to the README for this?

**Review by bambooch — CHANGES_REQUESTED** [review]
Please add a reference for how this should be used inside the project docs.

File: backend/controllers/email.controller.js (hunk 1 above)
Comment (Suvrat1629) [reply backend/controllers/email.controller.js:7]: Okk I have made the changes to the .env file in the README and added the recaptcha fields, do you also need me to write some sort of explanation or anything in the README for it?

Comment (bambooch) [reply backend/controllers/email.controller.js:7]: Hi yes please, if you can mayb add a bit more detail on where the key is generated and so on.

**Comment by bambooch** [comment]
Hi @Suvrat1629 I fail to understand how the captcha would work without its frontend counterpart.

Please take a look at the talk with gpt below and adapt the code accordingly 

https://chatgpt.com/share/67149310-32d4-8009-b31d-58acd41dbbb6

**Comment by Suvrat1629** [comment]
Hey @bambooch Sorry about that, I will work on the frontend but could you tell me where exactly do i need to add the frontend code because I am having a hard time navigating the code base and website.😅

**Comment by bambooch** [comment]
Hi @Suvrat1629 the captcha shouldn't work for the 'mailto' on the info@zeforge.ba email but on the form used for sending feedback that pops up on the modal. 

How to open the modal:

Click on the blue button in the footer below: 

image

Modal opens up:
image

The captcha should test the user before he can send the email successfully

Link to code: https://github.com/zeforgezenica/busez/blob/366b7ba2c9b9d60b27b1d7ac5655c0eb1c6ffabb/frontend/src/app/components/Footer.tsx#L116

**Pull request closed without merging**